doc/man/8/rbd.rst

```diff
@@ -568,19 +572,19 @@ Commands
 Image, snap, group and journal specs
 ====================================
 
-| *image-spec*      is [*pool-name*/]\ *image-name*
-| *snap-spec*       is [*pool-name*/]\ *image-name*\ @\ *snap-name*
-| *group-spec*      is [*pool-name*/]\ *group-name*
-| *group-snap-spec* is [*pool-name*/]\ *group-name*\ @\ *snap-name*
-| *journal-spec*    is [*pool-name*/]\ *journal-name*
+| *image-spec*      is [*pool-name*/[*namespace-name*/]]\ *image-name*
+| *snap-spec*       is [*pool-name*/[*namespace-name*/]]\ *image-name*\ @\ *snap-name*
+| *group-spec*      is [*pool-name*/[*namespace-name*/]]\ *group-name*
+| *group-snap-spec* is [*pool-name*/[*namespace-name*/]]\ *group-name*\ @\ *snap-name*
+| *journal-spec*    is [*pool-name*/[*namespace-name*/]]\ *journal-name*
```

Comment:
taking that a namespace is global across pools, wouldn't it be more correct to use "namespace-name/pool-name/..."?

Reply:
Namespaces are a just a hidden object prefix, so they are technically properties of the pool.  The ```rbd namespace create``` is tied to a pool, for example.

Reply:
Technically, yes. But then I would expect that something like this would work:
```
  rbd create ns1/pool/image --data-pool ns2/data_pool
```
Which does not.

When creating an image in a namespace, we expect that this namespace exists in matadata pool, data pool, journal pool, group pool, etc. So from this point of view it is global. And then the command `rbd namespace create` just means create/add the namespace into that pool.

Another way is not to extend the notation so  a namespace can be specified only via command line options. We already have problems with this notation, as it does not work e.g. for a pool or an image with slash in its name.

Actually, I am ok with any way. I just wanted to make sure you considered all options.